src/LaravelQueueRabbitMQServiceProvider.php

```diff
@@ -43,6 +44,8 @@ public function register(): void
                 );
             });
 
+            $this->app->bind(RabbitMQJobFactory::class);
```

Comment:
What do you think of making this configurable through connection config?

Reply:
Both alternatives are possible. In case of using the configuration connection, ´RabbitMQJobFactory.php´ would not be necessary and all this PR does not make sense. What you prefer is ok.
sorry, my english is very bad,

Reply:
My idea is to add `job_factory` config option with default value as `VladimirYuldashev\LaravelQueueRabbitMQ\Queue\Jobs\RabbitMQJobFactory::class`

Reply:
I think it's great. Will you do it? or do you want me to do a new PR?

Reply:
I think you can do it in this PR.

Reply:
```
        'rabbitmq' => [

            'driver' => 'rabbitmq',
            'queue' => env('RABBITMQ_QUEUE', 'default'),
            'connection' => PhpAmqpLib\Connection\AMQPLazyConnection::class,

            'hosts' => [
                [
                    'host' => env('RABBITMQ_HOST', '127.0.0.1'),
                    'port' => env('RABBITMQ_PORT', 5672),
                    'user' => env('RABBITMQ_USER', 'guest'),
                    'password' => env('RABBITMQ_PASSWORD', 'guest'),
                    'vhost' => env('RABBITMQ_VHOST', '/'),
                ],
            ],

            'options' => [
                'ssl_options' => [
                    'cafile' => env('RABBITMQ_SSL_CAFILE', null),
                    'local_cert' => env('RABBITMQ_SSL_LOCALCERT', null),
                    'local_key' => env('RABBITMQ_SSL_LOCALKEY', null),
                    'verify_peer' => env('RABBITMQ_SSL_VERIFY_PEER', true),
                    'passphrase' => env('RABBITMQ_SSL_PASSPHRASE', null),
                ],
                'queue' => [
                    'job_factory' => \VladimirYuldashev\LaravelQueueRabbitMQ\Queue\Jobs\RabbitMQJobFactory::class,
                ],
            ],

            /*
             * Set to "horizon" if you wish to use Laravel Horizon.
             */
            'worker' => env('RABBITMQ_WORKER', 'default'),

         ],
```

Reply:
Can i ask why you would prefer an factory in this case?
I think its over complicated.

When people want to do something else than the default `RabitMQJob::class`, they have to create multiple classes to extend the functionality.

I have an other perspective on this mather....

In the config just write
```
'options' => [
    ...
    
    'queue' => [
        ...

        'job' => \VladimirYuldashev\LaravelQueueRabbitMQ\Queue\Jobs\RabbitMQJob::class,
        
        ...
        ],
            
        ...
    ],

```

In the case of messages not created by laravel, and won't respect the payload... 

just write your own RabbitMQJob class. 

```php
<?php

namespace App\Queue\Jobs;

use VladimirYuldashev\LaravelQueueRabbitMQ\Queue\Jobs\RabbitMQJob as BaseJob;

class RabbitMQJob extends BaseJob
{

    /**
     * Fire the job.
     *
     * @return void
     */
    public function fire()
    {
        $payload = $this->payload();

        $class = WhatheverClassName::class;
        $method = 'handle';

        ($this->instance = $this->resolve($class))->{$method}($this, $payload);
    }
}

`````
What i would do is extend the current RabbitMQJob::class and overwrite the 'fire()' method.
The original fire method expects an payload with a key 'job' and it's value has to be an class that exists.
So by overwriting this method you can do watherver you want with custom payloads.

Reply:
Perfect!. We wait for the mergeo of PR [https://github.com/vyuldashev/laravel-queue-rabbitmq/pull/370](https://github.com/vyuldashev/laravel-queue-rabbitmq/pull/370)